File: [20190106-20190109]/angular/jeuxVideoFront/src/app/metier/jeux-video.ts
import { Editeur } from './editeur';
import { PlateForme } from './plate-forme';
import { Genre } from './genre';

export class JeuxVideo {
    constructor(public id : number,
                public nom: string,
                public dateSortie: Date,
                public editeur? : Editeur,
                public genres ? : Genre[],
                public plateFormes ? : PlateForme[]
                ) {}
}

File: [20190106-20190109]/java/nanomania-form/src/main/java/com/edugroupe/nanomaniaform/metier/JeuxVideo.java
package com.edugroupe.nanomaniaform.metier;

import java.time.LocalDate;
import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor
@Entity
public class JeuxVideo {
	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
						private int id;
						private String nom;
						private LocalDate dateSortie;
	@JsonIgnore
	@ManyToMany			private Set<Genre> genres;
	@JsonIgnore
	@ManyToMany 		private Set<PlateForme> plateFormes;
	@JsonIgnore
	@ManyToOne 			private Editeur editeur;
	
	
}
